src/libraries/System.Net.Connections/tests/System.Net.Connections.Tests/Sockets/SocketsConnectionFactoryTests.cs

```diff
@@ -225,36 +225,6 @@ public async Task Connection_Pipe_ReadWrite_Success(EndPoint endPoint, SocketTyp
             Assert.True(rr.Buffer.FirstSpan.SequenceEqual(sendData));
         }
 
-        [Fact]
-        public async Task Connection_Stream_FailingOperation_ThowsNetworkException()
-        {
-            using var server = SocketTestServer.SocketTestServerFactory(SocketImplementationType.Async, IPAddress.Loopback);
-            using SocketsConnectionFactory factory = new SocketsConnectionFactory(SocketType.Stream, ProtocolType.Tcp);
-            using Connection connection = await factory.ConnectAsync(server.EndPoint);
-
-            connection.ConnectionProperties.TryGet(out Socket socket);
-            Stream stream = connection.Stream;
-            socket.Dispose();
-
-            Assert.Throws<NetworkException>(() => stream.Read(new byte[1], 0, 1));
-            Assert.Throws<NetworkException>(() => stream.Write(new byte[1], 0, 1));
-        }
-
-        [Fact]
-        public async Task Connection_Pipe_FailingOperation_ThowsNetworkException()
-        {
-            using var server = SocketTestServer.SocketTestServerFactory(SocketImplementationType.Async, IPAddress.Loopback);
-            using SocketsConnectionFactory factory = new SocketsConnectionFactory(SocketType.Stream, ProtocolType.Tcp);
-            using Connection connection = await factory.ConnectAsync(server.EndPoint);
-
-            connection.ConnectionProperties.TryGet(out Socket socket);
-            IDuplexPipe pipe = connection.Pipe;
-            socket.Dispose();
-
-            await Assert.ThrowsAsync<NetworkException>(() =>  pipe.Input.ReadAsync().AsTask());
-            await Assert.ThrowsAsync<NetworkException>(() => pipe.Output.WriteAsync(new byte[1]).AsTask());
-        }
```

Comment:
If we are keeping the rest of the code anyways, can we preserve these tests by refactoring the assertions for `IOException`? I would prefer not to loose them.

Reply:
As far as I can tell, the only reason these tests exist is to check for NetworkException specifically. So I'm not sure what we get by checking for IOException.

Reply:
I agree that the assertion is less valuable in it's current form, but is it more likely that we are about to re-introduce `NetworkException` eventually? In that case keeping them around will ensure that we don't loose them by accident.

Reply:
Depending on the direction we go, we will make a branch from master that _reverts_ this PR and continue to iterate the API --so, I don't think we will lose anything.